core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2523,6 +2524,8 @@ private[spark] class CallerContext(
    taskId: Option[Long] = None,
    taskAttemptNumber: Option[Int] = None) extends Logging {
 
+   val upstreamCallerContextStr =
+     if (upstreamCallerContext.isDefined) s"_${upstreamCallerContext.get}" else ""
```

Comment:
`upstreamCallerContextStr` needs to combine some characters/words with `upstreamCallerContext.get`, so `getOrElse` is not used here and below.